back/app/serializers/web_api/v1/idea_serializer.rb

```diff
@@ -10,6 +10,11 @@ class WebApi::V1::IdeaSerializer < WebApi::V1::BaseSerializer
     TextImageService.new.render_data_images object, :body_multiloc
   end
 
+  attribute :custom_field_values do |object|
+    custom_field_values = CustomFieldService.remove_hidden_custom_fields(object.custom_field_values)
+    CustomFieldService.remove_disabled_custom_fields(custom_field_values)
```

Comment:
I would not use the same name for this variable as for the attribute that is being defined, because it is confusing.
```suggestion
    visible_field_values = CustomFieldService.remove_hidden_custom_fields(object.custom_field_values)
    CustomFieldService.remove_disabled_custom_fields(visible_field_values)
```